src/main/kotlin/com/openlattice/collections/processors/UpdateEntitySetCollectionMetadataProcessor.kt

```diff
@@ -0,0 +1,52 @@
+package com.openlattice.collections.processors
+
+import com.kryptnostic.rhizome.hazelcast.processors.AbstractRhizomeEntryProcessor
+import com.openlattice.collections.EntitySetCollection
+import com.openlattice.edm.requests.MetadataUpdate
+import java.util.*
+
+class UpdateEntitySetCollectionMetadataProcessor(val update: MetadataUpdate):
+        AbstractRhizomeEntryProcessor<UUID, EntitySetCollection, EntitySetCollection?>() {
+
+    override fun process(entry: MutableMap.MutableEntry<UUID, EntitySetCollection>?): EntitySetCollection? {
```

Comment:
The default generated signature here is wrong. It's a bug and it should be MutableMap.MutableEntry<UUID, EntitySetCollection?>). You won't ever get a null entry passed in (null keys aren't supported AFAIK without hax). You may occasionally end up with null keys.

Reply:
Is there an intellij setting or something that can fix the generated signature?

Reply:
To be more specific there is a bug in the generator since it is intellij generating from Java interface.

Reply:
I wish :-/